mobile/src/screens/partnerWelcome.js

```diff
@@ -33,7 +35,9 @@ class PartnerWelcomeScreen extends Component {
     if (this.props.settings.referralCode) {
       // We want to direct the user directly to dapp onboarding
       const url = new URL(this.props.settings.network.dappUrl)
-      url.hash = '/onboard'
+      url.hash =
+        '/onboard?referralCode=' +
+        encodeURIComponent(this.props.settings.referralCode)
```

Comment:
This has been tested to work with hash router?  `domain/path#hash?fakeParam=fakevar` is a non-standard URL and those aren't real GET params.

Reply:
It did work on an Android device.

Reply:
I'm going to have to take your word for this, but something to keep in mind since this is an invalid URL.